iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/execute/task/AbstractCompactionTask.java

```diff
@@ -372,21 +374,56 @@ public boolean isDiskSpaceCheckPassed() {
     return CompactionUtils.isDiskHasSpace();
   }
 
-  protected void validateTsFileResource(
-      List<TsFileResource> targetTsFileList, boolean needValidateOverlap) {
+  protected void validateCompactionResult(
+      List<TsFileResource> sourceSeqFiles,
+      List<TsFileResource> sourceUnseqFiles,
+      List<TsFileResource> targetFiles)
+      throws CompactionValidationFailedException {
+    // skip TsFileResource which is marked as DELETED status
+    List<TsFileResource> validTargetFiles =
+        targetFiles.stream().filter(resource -> !resource.isDeleted()).collect(Collectors.toList());
+    CompactionTaskType taskType = getCompactionTaskType();
+    boolean needToValidateTsFileCorrectness = taskType != CompactionTaskType.INSERTION;
```

Comment:
These **CompactionTaskType** are supposed to be specified by task new or another step, but I can't find a place to assign a value to the task, make sure the task corresponds to its task type